### ISSUE TITLE
`minMaxLoc` incompatible with some OpenCL implementations

### ISSUE THREAD

**Issue by JoeHowse** [opencv/opencv#22860]
### System Information

OpenCV Python version: 5.0.0-pre (commit [d976272](https://github.com/opencv/opencv/commit/d976272d23991873d0781f9ea3f6e61b070b134c))
Operating system / platform: Manjaro on Udoo Bolt v8

### Detailed description

On my system, when using `UMat` input, `minMaxLoc` outputs the following OpenCL compilation error and then the whole system locks up:

```
OpenCL program build log: core/minmaxloc
Status -11: CL_BUILD_PROGRAM_FAILURE
-D DEPTH_0 -D srcT1=uchar -D WGS=256 -D srcT=uchar4 -D WGS2_ALIGNED=128 -D DOUBLE_SUPPORT -D HAEV_SRC_CONT -D HAVE_MASK_CONT -D kercn=4 -D NEED_MAXVAL -D dstT1=uchar -D dstT=uchar4 -D convertToDT=noconvert -D wdepth=0 -D convertFromU=noconvert -D MINMAX_STRUCT_ALIGNMENT=8 -D AMD_DEVICE input.cl:8:1: error: OpenCL C version 1.1 does not support the 'static' storage class specifier
```

Based on the error message, I guess a similar problem would affect all systems running OpenCL 1.1.

Expected behaviour:  The function should fall back to a compatible CPU implementation if the system lacks the necessary features for the OpenCL-accelerated implementation.

### Steps to reproduce

Run `minMaxLoc` with `UMat` input on a system with OpenCL 1.1.

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by alalek** [comment]
BTW, location of the code is here: https://github.com/opencv/opencv/blame/4.6.0/modules/core/src/opencl/minmaxloc.cl#L16

**Comment by asmorkalov** [comment]
Merged to 3.4. Will be merged to 4.x and 5.x later during regular merge procedure.
